server/data/helper_party_api_key.ts

```diff
@@ -0,0 +1,53 @@
+"use server";
+
+import bcrypt from "bcrypt";
+import { createSupabaseServiceClient } from "@/data/supabase_server_service_client";
+
+export async function validateApiKey(
```

Comment:
that's certainly easier, though I'm less concerned about leaking anything. cryptographically, finding an expired or revoked key by brute force should be as hard as finding a valid key. 

relaying the detailed information to the helper party will also improve debugging. 

There is a possibility that this list gets large over time, potentially by revoking lots of keys (due to a bug or user error.) 

I suppose my bias is to lean towards developer (helper party) ergonomics, but I don't have a strong opinion.

Reply:
> that's certainly easier, though I'm less concerned about leaking anything. cryptographically, finding an expired or revoked key by brute force should be as hard as finding a valid key.

I don't think you can brute force a valid key this way, but you can learn that the key you possess is indeed valid, but maybe expired. So that leaks some information.

I don't object strongly either, but my preference would be to avoid leakages if at all possible

Reply:
fair enough, I'll update this to only unexpired and unrevoked keys.